### PR TITLE
implemented password strength checker

### PR DESCRIPTION
**Pull request by lroche823**
for Task 2 #523

### CODE REVIEW

File: src/Components/SignUpPassword.jsx [hunk 1]
```diff
@@ -44,13 +46,79 @@ export default SignUpPassword = (props) => {
       .done();
   }, []);
 
+  const onChangePassword = (password) => {
+    setPassword(password);
+    checkPasswordStrength(password);
+  };
+
+  const checkPasswordStrength = (password) => {
```
Comment (dannydaking23) [inline src/Components/SignUpPassword.jsx:54]: The checkPasswordStrength function fulfills all requisite requirements listed in Task 2 for checking the user's password strength.

File: src/Components/SignUpPassword.jsx [hunk 2]
```diff
@@ -44,13 +46,79 @@ export default SignUpPassword = (props) => {
...
+    let criteriaCount = 0;
+
```
Comment (dannydaking23) [inline src/Components/SignUpPassword.jsx:56]: The following if statements properly tally the different criteria included in the password.

File: src/Components/SignUpPassword.jsx [hunk 3]
```diff
@@ -44,13 +46,79 @@ export default SignUpPassword = (props) => {
...
+    }
+    // tests number
+    if (/[0-9]/.test(password)) {
+      criteriaCount++;
+    }
+    // tests symbol
+    if (/[ `!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?~]/.test(password)) {
+      criteriaCount++;
+    }
+
```
Comment (dannydaking23) [inline src/Components/SignUpPassword.jsx:73]: The following if else statements correctly assigned password strength based off of their length, different types of inputs, and whether they are included in the bad password list.

File: src/Components/SignUpPassword.jsx [hunk 4]
```diff
@@ -44,13 +46,79 @@ export default SignUpPassword = (props) => {
...
+      criteriaCount === 3 &&
+      password.length >= 5 &&
+      !badPasswords.includes(password)
+    ) {
+      setPasswordStrength(1);
+    } else {
+      setPasswordStrength(0);
+    }
+  };
+
```
Comment (dannydaking23) [inline src/Components/SignUpPassword.jsx:90]: The displayPasswordStrength function displays the different colors assigned with different strengths very well.

File: src/Components/SignUpPassword.jsx [hunk 5]
```diff
@@ -44,13 +46,79 @@ export default SignUpPassword = (props) => {
...
+      strengthMessage = 'Password Strength: Poor';
+    }
+
+    return (
+      <Text style={{color: textColor, textAlign: 'center'}}>
+        {strengthMessage}
+      </Text>
+    );
+  };
+
```
Comment (dannydaking23) [inline src/Components/SignUpPassword.jsx:112]: Edits to the onPress function below disallow user user to create a password if it has a length <= 4 or if its strength returns 0.

**Review by dannydaking23 — APPROVED** [review]
Changes to the files achieve the goals of the task and I approve the PR.

**Review by MiklBack — APPROVED** [review]
The changes to the files achieve the goals of the task and I approve the PR.

**Pull request closed without merging**